### ISSUE TITLE
/errorlog should handle valid directory/file names

### ISSUE THREAD

**Issue by duncanp-sonar** [dotnet/roslyn#59110]
**Version Used**:  csc/vbc 4.1.0-1.21551.6

The `/errorlog:` argument is expected to be in the form `<file>[,version={1|1.0|2|2.1}]`.

However, the argument is not parsed correctly if the `<file>` part contains a semi-colon or comma, both of which are valid (though uncommon) in directory and file names on Windows and Linux.

**Steps to Reproduce**:

1. Issue the command:

`csc Program.cs /errorlog:with,comma.log`

Same for `vbc`.

**Expected Behavior**:
The compilation succeeds and the error log is produced.

**Actual Behavior**:
The compilation fails with a message like the following:

>error CS2046 : Command-line syntax error : 'with,comma.log' is not a valid value for the '/errorlog:' option. The value must be of the form '<file>[,version={1|1.0|2|2.1}]'. 

**Additional notes**
In practice, it's unlikely users will pass file names with commas or semi-colons directly. We [encountered the problem](https://community.sonarsource.com/t/error-in-sonarqube-while-executing-dotnet-build/56556) because the targets in [our third-party tool](https://docs.sonarqube.org/latest/analysis/scan/sonarscanner-for-msbuild/) set the `ErrorLog` to an absolute path, and a user happened to run their build under a directory containing a comma in the name.

**Comment by Youssef1313** [comment]
@duncanp-sonar Can you try enclosing the file path with double quotes?

**Comment by duncanp-sonar** [comment]
@Youssef1313 sure. I tried a number of variations, all of which fail (the screen shot below is for `vbc` - 'csc' fails the same way).

![image](https://user-images.githubusercontent.com/10757559/151356238-cfeab08e-0094-491e-9873-ffb792b2dc07.png)

**Comment by Youssef1313** [comment]
Yeah it's expected that csc and vbc behave the same because the code is shared in the `Core` layer.

https://github.com/dotnet/roslyn/blob/3b32fc36ec4c2a81457d90f965c89c5b6fc37082/src/Compilers/Core/Portable/CommandLine/CommandLineParser.cs#L204-L260

I *think* it's a bug, it should have worked with double quotes.

**Comment by Youssef1313** [comment]
Quick observation, the double quote appears to be omitted/eaten somewhere.
